staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -17,55 +17,128 @@ limitations under the License.
+	// Example value: "certificates.cert-manager.io,configmaps,deployments.apps,secrets,services"
+	ApplySetGRsAnnotation = "applyset.k8s.io/contains-group-resources"
+
+	// ApplySetParentIDLabel is the key of the label that makes object an ApplySet parent object.
+	// Its value MUST be the base64 encoding of the hash of the GKNN of the object it is on,
+	// in the form base64(sha256(<name>.<namespace>.<kind>.<group>)), using the URL safe encoding of RFC4648.
+	ApplySetParentIDLabel = "applyset.k8s.io/id"
+
+	// ApplysetPartOfLabel is the key of the label which indicates that the object is a member of an ApplySet.
+	// The value of the label MUST match the value of ApplySetParentIDLabel on the parent object.
+	ApplysetPartOfLabel = "applyset.k8s.io/part-of"
 )
 
 var defaultApplySetParentGVR = schema.GroupVersionResource{Version: "v1", Resource: "secrets"}
 
 // ApplySet tracks the information about an applyset apply/prune
 type ApplySet struct {
-	// ParentRef is the reference to the parent object that is used to track the applyset.
-	ParentRef *ApplySetParentRef
+	// parent is the reference to the parent object that is used to track the applyset.
+	parent *ApplySetParent
 
-	// resources is the set of all the resources that (might) be part of this applyset.
-	resources map[schema.GroupVersionResource]struct{}
+	// toolingID is the value to be used and validated in the applyset.k8s.io/tooling annotation.
+	toolingID ApplySetTooling
 
-	// namespaces is the set of all namespaces that (might) contain objects that are part of this applyset.
-	namespaces map[string]struct{}
+	// currentResources is the set of resources that are part of the sever-side set as of when the current operation started.
+	currentResources map[schema.GroupVersionResource]*meta.RESTMapping
+
+	// currentNamespaces is the set of namespaces that contain objects in this applyset as of when the current operation started.
+	currentNamespaces sets.Set[string]
+
+	// updatedResources is the set of resources that will be part of the set as of when the current operation completes.
+	updatedResources map[schema.GroupVersionResource]*meta.RESTMapping
+
+	// updatedNamespaces is the set of namespaces that will contain objects in this applyset as of when the current operation completes.
+	updatedNamespaces sets.Set[string]
+
+	restMapper meta.RESTMapper
+
+	// client is a client specific to the parent's type
+	client resource.RESTClient
 }
 
 var builtinApplySetParentGVRs = map[schema.GroupVersionResource]bool{
 	defaultApplySetParentGVR:                true,
 	{Version: "v1", Resource: "configmaps"}: true,
 }
 
-// ApplySetParentRef stores object and type meta for the parent object that is used to track the applyset.
-type ApplySetParentRef struct {
-	Name        string
-	Namespace   string
-	RESTMapping *meta.RESTMapping
+// ApplySetParent stores object and type meta for the parent object that is used to track the applyset.
+type ApplySetParent struct {
```

Comment:
Nit: probably leaving the previous name `ApplySetParentRef` was more appropriate, since you don't keep the object, but only a reference to it.